### PR TITLE
Improvements to ArcGisMapServerImageryProvider.

### PR DESCRIPTION
**Pull request by kring**
* Add support for automatic raster feature picking.
* Add the ability to specify the desired tiling scheme, rectangle, and width and height of tiles to the constructor.
* Add the ability to access dynamic ArcGIS MapServer layers by specifying the `layers` parameter to the constructor.
* Fix a bug that could cause incorrect rendering with a "singleFusedMapCache" in the geographic projection (EPSG:4326).

### CODE REVIEW

File: Source/Scene/ArcGisMapServerImageryProvider.js [hunk 1]
```diff
@@ -500,10 +557,69 @@ define([
      * @return {Promise} A promise for the picked features that will resolve when the asynchronous
      *                   picking completes.  The resolved value is an array of {@link ImageryLayerFeatureInfo}
      *                   instances.  The array may be empty if no features are found at the given location.
-     *                   It may also be undefined if picking is not supported.
+     *
+     * @exception {DeveloperError} <code>pickFeatures</code> must not be called before the imagery provider is ready.
      */
-    ArcGisMapServerImageryProvider.prototype.pickFeatures = function() {
-        return undefined;
+    ArcGisMapServerImageryProvider.prototype.pickFeatures = function(x, y, level, longitude, latitude) {
```
Comment (pjcozzi) [inline Source/Scene/ArcGisMapServerImageryProvider.js]: Do we have tests for this?

**Comment by pjcozzi** [comment]
Is it worth adding/extending a Sandcastle example for this?

**Comment by pjcozzi** [comment]
@kring do you expect this to make 1.8 on Wednesday?

**Comment by kring** [comment]
Yes, ideally.  Will get back to it Monday.

**Comment by kring** [comment]
This is ready.  I added a Sandcastle example and some tests.

**Comment by pjcozzi** [comment]
Thanks.

**Pull request merged**